### PR TITLE
Update libraries and tools

### PR DESCRIPTION
**Pull request by thePanz**
- [x] Update to symfony/console 5.4
- [x] Update to php-cs-fixer v3.x
- [x] Update to phpstan v1.9
- [x] Update to box/box v4.2
- [x] Fixed GuzzleHTTP security issues (suing v.6.5.8)
- [x] Support for php v7.4 and v8.x

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
 	composer install --optimize-autoloader --no-dev --no-suggest --quiet
-	./tools/box compile --quiet
+	./tools/box.phar compile --quiet
 
-tools/box:
-	wget --directory-prefix=tools --quiet https://github.com/humbug/box/releases/download/3.1.2/box.phar
-	mv tools/box.phar tools/box
-	chmod +x tools/box
+tools/box.phar:
+	wget --directory-prefix=tools --quiet https://github.com/humbug/box/releases/download/3.16.0/box.phar
```
Comment (dbu) [inline Makefile:9]: i downgraded to version 3 again as version 4 only works with php ^8.1

i think until everything is on php 8.1 we should still support 7.4.

Comment (dbu) [reply Makefile:9]: i verified that roger-q built on php 7.4 works runs fine in php 8.2

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,5 @@
+parameters:
+  level: 8
```
Comment (mjanser) [inline phpstan.neon:2]: Did you try level 9?

File: composer.json [hunk 3]
```diff
@@ -5,18 +5,21 @@
...
-        "php": "^7.1",
-        "enqueue/amqp-bunny": "^0.8.40",
-        "guzzlehttp/guzzle": "^6.3",
-        "pnz/json-exception": "^0.1.0",
-        "symfony/console": "^4.1"
+        "php": "^7.4 || ^8.0",
+        "enqueue/amqp-bunny": "^0.10.9",
+        "guzzlehttp/guzzle": "^6.5.8",
+        "pnz/json-exception": "^1.0.0",
+        "symfony/console": "^5.4"
```
Comment (mjanser) [inline composer.json:12]: Would it be possible to allow version 6 here as well, e.g. `^5.4 || ^6.0`?

Comment (mjanser) [reply composer.json:12]: Same for Guzzle , I guess.

File: Makefile [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
+	wget --directory-prefix=tools --quiet https://github.com/phpstan/phpstan/releases/download/1.9.17/phpstan.phar
+	chmod +x tools/phpstan.phar
 
-phpcs: tools/php-cs-fixer tools/phpstan
+phpcs: tools/php-cs-fixer.phar tools/phpstan.phar
 	composer install --optimize-autoloader --no-dev --no-suggest --quiet
-	tools/php-cs-fixer fix --dry-run --stop-on-violation -v
-	tools/phpstan analyze --level=7 --no-progress bin/ src/
+	tools/php-cs-fixer.phar fix --dry-run --stop-on-violation -v
+	tools/phpstan.phar analyze --level=8 --no-progress bin/ src/
```
Comment (mjanser) [inline Makefile]: With `phpstan.neon` you don't need to specify the options here.

File: composer.json (hunk 3 above)
Comment (dbu) [reply composer.json:12]: my thinking was that we build and deploy for php 7.4 and symfony 6 does not support php 7 anymore. 

but we could still allow it for people checking out the code and building the phar themselves on php 8.

File: phpstan.neon (hunk 2 above)
Comment (dbu) [reply phpstan.neon:2]: then we get a bunch of errors, i would rather not address that in the same pull request.

File: composer.json (hunk 3 above)
Comment (dbu) [reply composer.json:12]: guzzle would probably work, but i'd rather keep the change to a separate pull request to not change too much at the same time.

File: phpstan.neon (hunk 2 above)
Comment (mjanser) [reply phpstan.neon:2]: Yes, i agree.

**Pull request merged**